code/modules/modular_computers/hardware/_hardware.dm

```diff
@@ -87,3 +87,32 @@
 	damage += round(amount) 					// We want nice rounded numbers here.
 	damage = between(0, damage, max_damage)		// Clamp the value.
 
+// This is called whenever hardware is inserted into a modular computer.
+// Currently only has a use for 'manual' flash drives
+/obj/item/weapon/computer_hardware/proc/installed(var/obj/item/modular_computer/M)
+	if(istype(src, /obj/item/weapon/computer_hardware/hard_drive/portable/manual))
+		//Define the src as a portable flash drive
+		var/obj/item/weapon/computer_hardware/hard_drive/portable/flash = src
+		//Find the word processor
+		var/datum/computer_file/program/wordprocessor/WP = M.hard_drive.find_file_by_name("wordprocessor")
+		//List of added programs
+		var/list/addedProgs = list()
+		//Find all stored guides in the flash drive and transfer them over
+		for(var/datum/computer_file/data/text/manual/F in flash.stored_files)
+			//Don't add unnecessary copies
+			if(!M.hard_drive.stored_files.Find(F))
+				M.hard_drive.store_file(F)
+			addedProgs.Add(F.filename)
+		//Reset the word processors data in case anything is currently inside it.
+		//This generates a 'changelog' to show whats been added
+		WP.open_file = "Added files"
```

Comment:
This seems oddly involved. isn't there a simple way to just tell the computer to load a certain file and let it pick what program to use?